lib/utmp.c

```diff
@@ -160,8 +160,7 @@ get_current_utmp(void)
 		    && (USER_PROCESS  != ut->ut_type))
 			continue; /* Look for 'login' or 'user' entries only */
 
-		if (   (   (ut->ut_pid == getpid ())
-		        || (ut->ut_pid == getppid ()))
+		if (   (main_pid == ut->ut_pid)
 		    && ('\0' != ut->ut_id[0])
 		    /* A process may have failed to close an entry
 		     * Check if this entry refers to the current tty */
```

Comment:
Would you mind separating the first 3 commits from the last two?  I'll merge the first three soon, and then we can review the other two separately.

Reply:
Yeah, you could keep this PR complete, and then add a new PR with the first 3 commits.  I'll apply that one first, and then this one can be rebased on top of it.